HCIDOCS-316: Add information about Metal3 database to bare-metal IPI troubleshooting (#81501)
openshift/openshift-docs: modules/bmo-troubleshooting-latency-when-deleting-a-baremetalhost-resource.adoc

// Module included in the following assemblies:
// //installing/installing_bare_metal_ipi/installing_bare_metal_ipi/ipi-install-troubleshooting.adoc

:_mod-docs-content-type: PROCEDURE
[id="ipi-install-troubleshooothing-latency-when-deleting-a-baremetalhost-resource_{context}"]

= Troubleshooting latency when deleting a BareMetalHost resource

> When the Bare Metal Operator (BMO) deletes a `BareMetalHost` resource, Ironic deprovisions the bare-metal host with a process known as cleaning. If cleaning does not succeed, deleting the `BareMetalHost` resource takes considerable time and may not fully delete. The deleting process retries three times.

Reviewer: ```suggestion
When the Bare Metal Operator (BMO) deletes a `BareMetalHost` resource, Ironic deprovisions the bare-metal host with a process called cleaning. When cleaning fails, the deletion of the `BareMetalHost` resource by the BMO takes considerable time and might not succeed. The deleting process retries three times.
```
"might" instead of "may" for translatability
a little unclear to me who then is deleting things if the cleaning fails; implies that a user might go in and try manually but it takes a long time and might not work

Author: Actually, the problem is a little bit different. Users get impatient when it takes to long and they try to manually override too soon. So a user might go in and try manually, and create problems. So we're trying to steer them in the right direction about it taking time without overloading them with internal implementation details of what's happening. People were using a workaround found here https://bugzilla.redhat.com/show_bug.cgi?id=1837505#c15, which isn't a good idea; namely, they should disable cleaning rather than removing finalizers.

[IMPORTANT]
====

> Do not remove finalizers from `BareMetalHost` resources.

Reviewer: You had a longer note in the QE approved PR, https://github.com/openshift/openshift-docs/pull/80775/files#diff-44ef2609fd9567f86debc306604b0dadb4b9e5b8dfbea87cd267342c754e8f58R20

> When the Bare Metal Operator (BMO) deletes a `BareMetalHost` resource, Ironic deprovisions the bare-metal host. For example, this might happen when scaling down a machine set. Deprovisioning involves a process known as "cleaning". If cleaning doesn't succeed, the deletion of the `BareMetalHost` resource will take a long time, because it retries three times after a timeout. The deletion might not finish.

Reviewer 2: ```suggestion
When the Bare Metal Operator (BMO) deletes a `BareMetalHost` resource, Ironic deprovisions the bare-metal host with a process known as cleaning. If cleaning does not succeed, deleting the `BareMetalHost` resource takes considerable time and may not fully delete. The deleting process retries three times.
```
Would something like this work? I think the information is great; it just needs to be cleaned up a little and simplified for the reader.

> Do not remove finalizers from `BareMetalHost` resources that take a long time to get deleted.

Reviewer 2: ```suggestion
Do not remove finalizers from `BareMetalHost` resources before deletion.
```
Again, not sure if this says the same thing. I'm trying to keep the note as clear as possible.

Author: That's slightly misleading. So I just removed the "before deletion" part.
